Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,248 @@
+  }
+  this._light.blendMode = PIXI.BLEND_MODES.ADD;
+
+  if (this.DEBUG) {
+    if (this._graphics === undefined) {
+      this._graphics = new PIXI.Graphics();
+      this._graphics
+        .lineStyle(1, 0xff0000, 1)
+        .moveTo(this._object.x, this._object.y)
+        .lineTo(this._object.x - this._radius, this._object.y + this._radius)
+        .moveTo(this._object.x, this._object.y)
+        .lineTo(this._object.x + this._radius, this._object.y + this._radius)
+        .moveTo(this._object.x, this._object.y)
+        .lineTo(this._object.x + this._radius, this._object.y - this._radius)
+        .moveTo(this._object.x, this._object.y)
+        .lineTo(this._object.x - this._radius, this._object.y - this._radius);
+    }
+    if (this._debugLight === undefined) {
+      this._debugLight = new PIXI.Container();
+      this._debugLight.addChild(this._light);
+      this._debugLight.addChild(this._graphics);
+    }
+  }
+};
+
+gdjs.LightRuntimeObjectRenderer = gdjs.LightRuntimeObjectPixiRenderer; //Register the class to let the engine use it.
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.getRendererObject = function () {
+  // Mandatory, return the internal PIXI object used for your object:
+  if (this.DEBUG) {
+    return this._debugLight;
+  }
+  return this._light;
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.ensureUpToDate = function () {
+  if (this.DEBUG) this.updateGraphics();
+  this.updateVertexBuffer();
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.updateGraphics = function () {
+  var vertices = [this._object.x, this._object.y];
+  var raycast = this.raycastTest().flat(1);
+  vertices.push.apply(vertices, raycast);
+
+  this._graphics.clear();
+  for (var i = 0; i < vertices.length; i += 2) {
+    var lineColor = i % 4 === 0 ? 0xff0000 : 0x00ff00;
+    this._graphics
+      .lineStyle(1, lineColor, 1)
+      .moveTo(vertices[0], vertices[1])
+      .lineTo(vertices[i], vertices[i + 1]);
+    if (i !== vertices.length - 2) {
+      this._graphics.lineTo(vertices[i + 2], vertices[i + 3]);
+    } else {
+      this._graphics.lineTo(vertices[2], vertices[3]);
+    }
+  }
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.updateVertexBuffer = function () {
+  this._light.shader.uniforms.center = new Float32Array([
+    this._object.x,
+    this._object.y,
+  ]);
+
+  var raycast = this.raycastTest();
+
+  var vertexBuffer = [this._object.x, this._object.y];
+
+  for (var i = 2; i < 2 * raycast.length + 2; i += 2) {
+    vertexBuffer[i] = raycast[i/2 - 1][0];
+    vertexBuffer[i + 1] = raycast[i/2 - 1][1];
+  }
+
+  var indexBuffer = [];
+
+  for (var i = 0; i < 3 * raycast.length; i += 3) {
+    indexBuffer[i] = 0;
+    indexBuffer[i + 1] = i / 3 + 1;
+    if ((i / 3) + 1  !== raycast.length) indexBuffer[i + 2] = i / 3 + 2;
+    else indexBuffer[i + 2] = 1;
+  }
+
+  this._light.geometry
+    .getBuffer('aVertexPosition')
+    .update(new Float32Array(vertexBuffer));
```

Comment:
@4ian is there a way to use only one instance of `Float32Array` and updating it, instead of creating a new instance every time? I'm not familiar with "updating" typed arrays. 😅 Maybe we could then use `this._vertexBuffer` and `this._indexBuffer`?

Reply:
Yes totally! :) I think we should store the vertexBuffer and the indexBuffer as properties of the object. 
Float32Array are not resizable, which is I think still kind of fine. We can:
- set _vertexBuffer to null at the creation of the object
- when the vertices are updated, either:
  - the new vertices have the same size as before (will be the case in 99% of cases, because even if an object is moving, most of the time it will stay near the light, and then when it's gone outside of the light radius, well it's gone!), in which case you don't recreate the Float32Array but update it.
  - or the new vertices length is different, in which case you recreate a new Float32Array (there might be some optimisation like keeping the same array if it's not too large, and just ignore the last vertices in the shader??).

Same idea for indexBuffer - we should strive to avoid allocations and reuse them as possible. No reason we have to allocate them at every frame (it's for sure costly when we have lots of lights)